modal-go/image.go

```diff
@@ -4,18 +4,40 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"strings"
 
 	pb "github.com/modal-labs/libmodal/modal-go/proto/modal_proto"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
+// ImageDockerfileCommandsOptions are options for Image.DockerfileCommands().
+type ImageDockerfileCommandsOptions struct {
+	// Secrets that will be made available to this layer's build environment.
+	Secrets []*Secret
+
+	// GPU reservation for this layer's build environment (e.g. "A100", "T4:2", "A100-80GB:4").
+	GPU string
```

Comment:
Typically, I would not include include GPU unless there is a request for it, but this seems simple enough to add.